lib/core/test/unit/Cardano/Wallet/DB/StateMachine.hs

```diff
@@ -526,6 +537,22 @@ generator (Model _ wids) = Just $ frequency $ fmap (fmap At) <$> concat
     genPrivKey :: Gen MPrivKey
     genPrivKey = elements ["pk1", "pk2", "pk3"]
 
+    genSortDir :: Gen SortOrder
+    genSortDir = QC.elements [Ascending, Descending]
+
+    genFilter :: Gen (Filter SlotId)
+    genFilter = Filter <$> genSId <*> genSId
+      where
+        genSId :: Gen (Maybe SlotId)
+        genSId = QC.oneof
+            [ return Nothing
+            , Just <$> (SlotId <$> genInt <*> genInt)
+            ]
+        -- FIXME: Temporary to check if this is tha cause of StateMachine
+        -- failure
+        genInt :: Num a => Gen a
+        genInt = fromIntegral <$> arbitrary @Word32
```

Comment:
Seems we need mind the ~48 bit limit on epochs from this hack https://github.com/input-output-hk/cardano-wallet/blob/29c9a6a9111d80d4874a63c07c2f138dfadae104/lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs#L231-L236